### DISCUSSION TITLE
How to understand the backtest postions report?

### DISCUSSION THREAD

**Discussion by graceyangfan** [nautechsystems/nautilus_trader#708] [Q&A] [answered]
In crypto_ema_cross_ethusdt_trade_ticks.py 
(1)Why the positionsEvent report give all postion the same ts_opened  ?
(2)some positionEvent  give the same avg_px_open,from my point of view,only reducing a postion,the avg_px_open keeps the same,but I find that some events are postion opened.

**Comment by cjdsellers** [comment]
Looking at the reports, I think you may have uncovered an issue with the position snapshots. The `avg_px_open` doesn't look correct, along with a few other fields.

I'll have a closer look tomorrow.

**Comment by cjdsellers** [comment, marked as answer]
Now fixed from c3abc8893547fe90bb48950eea6e0e15e07e93cf and as per https://github.com/nautechsystems/nautilus_trader/issues/719
